### PR TITLE
Enable simple shuffling with react routing

### PR DESCRIPTION
**Pull request by fransaycon**
Create the following core functionalities
1. Create a shuffled entries array and showcase them.
2. Navigation between the shuffled array
3. Routing capabilities. You may access through a single link making it the first by default. Accessing with no pathnames, or index only results in getting the first entry in the shuffled array.

### CODE REVIEW

File: src/ContributorDetails.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { useLocation } from "react-router-dom"
+
+const getCurrentSlug = (location) => {
+    const paths = location.pathname.split("/")
+
+    if(paths.length > 0){
+        return paths[1]
+    }    
+
+    return null
```
Comment (johndavedecano) [inline src/ContributorDetails.js]:
```js
return paths.length ? paths[0] : null
```
or maybe you want it like

```js
return paths.length > 1 ? paths[1] : null
```

File: src/index.js [hunk 2]
```diff
@@ -1,17 +1,38 @@
...
-ReactDOM.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>,
-  document.getElementById('root')
-);
+const generateRoutes = (entries) => {
+    if(entries.length === 0){
+        return null
+    }
```
Comment (johndavedecano) [inline src/index.js:10]:
```js
if (!entries.length) return null
```

File: src/index.js [hunk 3]
```diff
@@ -1,17 +1,38 @@
...
+    return [
+        <Route path="/" key="default" exact>
+            {entries[0].component}
+        </Route>,
+        ...entries.map((entry) => (
+            <Route path={`/${entry.slug}`} key={entry.slug}>
+                {entry.component}
+            </Route>
+        ))
+    ]
```
Comment (johndavedecano) [inline src/index.js]:
```js
const [initialRoute, ...otherRoutes] = entries

return [
    <Route path="/" key="default" component={initialRoute.component} exact />,
     ...otherRoutes.map((entry) => (
        <Route key={entry.slug} path={`/${entry.slug}`}  component={entry.component} />
      ))
    ]
```

Comment (fransaycon) [reply src/index.js]: I'll do the component part. The logic intended for this is to add the first as the default while still making the slug of the first one available as a route.

File: src/index.js (hunk 2 above)
Comment (fransaycon) [reply src/index.js:10]: better if include the typing check with ===. since this is not typed.

**Review by carldegs — APPROVED** [review]

**Review by johndavedecano — APPROVED** [review]

**Pull request merged**